game/view/transmission.lua

```diff
@@ -46,23 +59,39 @@ end
 function Transmission:draw()
   if self.warmup > 0 then return end
   local g = love.graphics
-  local ori_x = self.start.x + self.ori_ox
-  local ori_y = self.start.y + self.ori_oy
-  local tar_x = self.finish.x + self.tar_ox
-  local tar_y = self.finish.y + self.tar_oy
+  local ori = vec2(self.start.x + self.ori_ox, self.start.y + self.ori_oy)
+  local tar = vec2(self.finish.x + self.tar_ox, self.finish.y + self.tar_oy)
 
-  --Draw origin and target circle
+  --Draw origin and target circle--
   g.setColor(self.color[1]*_COLOR_DISTORTION,
              self.color[2]*_COLOR_DISTORTION,
              self.color[3]*_COLOR_DISTORTION,
              self.color[4])
-  g.circle("fill", ori_x, ori_y, _RADIUS * self.width_scale)
-  g.circle("fill", tar_x, tar_y, _RADIUS * self.width_scale)
+  g.circle("fill", ori.x, ori.y, _RADIUS * self.width_scale)
+  g.circle("fill", tar.x, tar.y, _RADIUS * self.width_scale)
 
-  --Draw line
+  --Draw line--
   g.setLineWidth(_WIDTH * self.width_scale)
   g.setColor(self.color)
-  g.line(ori_x, ori_y, tar_x, tar_y)
+  --Insert origin position
+  local points = {ori.x, ori.y}
+  --Insert bendings
+  local norm = (tar-ori):normalize()
+  local dist = ori:dist(tar)/(self.bending_number+1)
+  local sign = 1
+  for i = 1, self.bending_number do
+    local off = self.bending_offsets[i]
+    local angle = self.bending_angles[i]
+    local pos = ori+norm*dist*i
+    pos = pos + norm:perpendicular():rotate(angle)*off*sign
+    table.insert(points,pos.x)
+    table.insert(points,pos.y)
+    sign = sign*-1
+  end
+  --Insert target position
+  table.insert(points, tar.x)
+  table.insert(points, tar.y)
+  g.line(unpack(points))
```

Comment:
não é bom usar unpack para muitos atributos pq tem um limite. o graphics.line aceita uma tabela de pontos.

Reply:
Uhh forgot about that